src/parser.rs

```diff
@@ -596,7 +596,12 @@ impl CommentParser<&mut Revisioned> {
             ),
             Some('^') => {
                 let offset = pattern.chars().take_while(|&c| c == '^').count();
-                (self.line - offset, &pattern[offset..])
+                if let Some(match_line) = self.line.checked_sub(offset) {
+                    (match_line, &pattern[offset..])
+                } else {
+                    self.error(format!("//~^ pattern is trying to refer to {} lines previous, but is only on line {}", offset, self.line));
```

Comment:
```suggestion
                    self.error(format!("//~^ pattern is trying to refer to {} lines above, but the file only has {} lines", offset, self.line));
```

Reply:
"above" is better, but the fact that it's the currently processed line matters, to me "the file only has {} lines" reads as "the whole file", even in context. how about "trying to refer to {} lines above, but there are only {} lines above it"?

Reply:
Oh yea 🤦 that's much better